Common/AnatupleProducer.py

```diff
@@ -122,29 +126,30 @@ def createAnatuple(inFile, outFile, config, sample_name, snapshotOptions,range,
     if len(evtIds) > 0:
         df = df.Filter(f"static const std::set<ULong64_t> evts = {{ {evtIds} }}; return evts.count(event) > 0;")
     df = Baseline.applyMETFlags(df, config["GLOBAL"]["MET_flags"])
-    df = DefineAndAppend(df,"sample_type", f"static_cast<int>(SampleType::{config[sample_name]['sampleType']})")
-    df = DefineAndAppend(df,"period", f"static_cast<int>(Period::{period})")
-    df = DefineAndAppend(df,"X_mass", f"static_cast<int>({mass})")
+    df = df.Define("sample_type", f"static_cast<int>(SampleType::{config[sample_name]['sampleType']})")
+    df = df.Define("period", f"static_cast<int>(Period::{period})")
+    df = df.Define("X_mass", f"static_cast<int>({mass})")
     is_data = 'true' if isData else 'false'
-    df = DefineAndAppend(df,"is_data", is_data)
+    df = df.Define("isData", is_data)
 
     df = Baseline.CreateRecoP4(df)
     df = Baseline.DefineGenObjects(df, isData=isData, isHH=isHH)
     if isData:
         syst_dict = { 'nano' : 'Central' }
     else:
         df, syst_dict = Corrections.applyScaleUncertainties(df)
-    df,weight_branches = Corrections.getNormalisationCorrections(df, config, sample_name)
-    for br in weight_branches:
-        br_name = f'weight_{br}' if br != "Central" else "weight"
-        colToSave.append(br_name)
     for syst_name, source_name in syst_dict.items():
-        suffix = '' if syst_name in [ 'Central', 'nano' ] else f'_{syst_name}'
+        is_central = syst_name in [ 'Central', 'nano' ]
+        suffix = '' if is_central else f'_{syst_name}'
         if len(suffix) and not store_noncentral: continue
-        df_syst = addAllVariables(df, syst_name, isData, trigger_class)
+        dfw = DataFrameWrapper(df)
+        addAllVariables(dfw, syst_name, isData, trigger_class)
+        df_syst = dfw.df
+        df_syst, weight_branches = Corrections.getNormalisationCorrections(df_syst, config, sample_name, is_central)
```

Comment:
```suggestion
        weight_branches = dfw.Apply(Corrections.getNormalisationCorrections, config, sample_name, is_central and compute_unc_variations)
```
add `compute_unc_variations` as a command line argument

Reply:
ok